File: Data Structures & Algorithms/daily-temperatures/submission-0.go
func dailyTemperatures(temperatures []int) []int {
    htStack, res := []int{}, []int{}
    for _, temp := range temperatures {
        res = append(res, 0)
        // htStack
        idx := len(htStack)
        zc := 0
        for i := 1; i <= idx; i++ {
            // keep popping until 
            if htStack[idx-i] < temp {
                htStack = htStack[:idx-i]
                zc++
            } else {
                break
            }
        }
        htStack = append(htStack, temp)
        r := len(res) - 1
        for i := 0; i < zc; i++ {
            for  {
               r -= 1
               if res[r] == 0 {
                    res[r] = len(res) - r - 1
                    break
               }
            }
        }
    }

    return res
}

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        dp = [[0] * len(s) for _ in s]
        
        def check(x, y):
            if x == y:
                dp[x][y] = 1 
                return True
            elif x+1 == y:
                r = s[x] == s[y]
                if r: dp[x][y] = 1
                else: dp[x][y] = -1
                return r
            else:
                if s[x] != s[y]:
                    dp[x][y] = -1
                    return False
                if dp[x][y] != 0:
                    return dp[x][y] == 1
                dp[x][y] = check(x+1, y-1)
                return dp[x][y]
                 
        res = 0
        for st in range(0, len(s)):
            for l in range(0, len(s)):
                if st + l >= len(s): break
                if check(st, st+l): res += 1 

        return res